library/ssl_cli.c

```diff
@@ -258,27 +291,28 @@ static size_t ssl_get_ec_curve_list_length( mbedtls_ssl_context *ssl )
     return( ec_list_len );
 }
 
-static void ssl_write_supported_elliptic_curves_ext( mbedtls_ssl_context *ssl,
-                                                     unsigned char *buf,
-                                                     size_t *olen )
+static int ssl_write_supported_elliptic_curves_ext( mbedtls_ssl_context *ssl,
+                                                    unsigned char *buf,
+                                                    const unsigned char *end,
+                                                    size_t *olen )
 {
     unsigned char *p = buf;
-    const unsigned char *end = ssl->out_msg + MBEDTLS_SSL_OUT_CONTENT_LEN;
     size_t elliptic_curve_len = 0;
 
     *olen = 0;
 
-    MBEDTLS_SSL_DEBUG_MSG( 3, ( "client hello, adding supported_elliptic_curves extension" ) );
+    MBEDTLS_SSL_DEBUG_MSG( 3,
+        ( "client hello, adding supported_elliptic_curves extension" ) );
 
     /* Each elliptic curve is encoded in 2 bytes. */
     elliptic_curve_len = 2 * ssl_get_ec_curve_list_length( ssl );
     if( elliptic_curve_len == 0 )
-        return;
+        return( MBEDTLS_ERR_SSL_BAD_CONFIG );
 
     if( end < p || (size_t)( end - p ) < 6 + elliptic_curve_len )
     {
         MBEDTLS_SSL_DEBUG_MSG( 1, ( "buffer too small" ) );
-        return;
+        return( MBEDTLS_ERR_SSL_BAD_CONFIG );
```

Comment:
Why you didn't want to use the check buf macro in this place ?
 MBEDTLS_SSL_CHK_BUF_PTR( p, end, 6 + elliptic_curve_len );

Reply:
I didn't want to modify the code as much, and I wanted to stick to `MBEDTLS_ERR_SSL_BAD_CONFIG`. 
Using `MBEDTLS_SSL_CHK_BUF_PTR` would mean that there's no debug message (and maybe some tests depend on it), and the return code would be different (`MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL`).